src/box/lua/console.c

```diff
@@ -471,7 +471,17 @@ lbox_console_format_yaml(struct lua_State *L)
 	}
 	lua_replace(L, 1);
 	lua_settop(L, 1);
-	return lua_yaml_encode(L, serializer_yaml, NULL, NULL);
+	int ret = lua_yaml_encode(L, serializer_yaml, NULL, NULL);
+	if (ret == 2) {
+		/*
+		 * Nil and error object are pushed onto the stack.
+		 */
+		assert(lua_isnil(L, -2));
+		assert(lua_isstring(L, -1));
+		return luaL_error(L, lua_tostring(L, -1));
+	}
+	assert(ret == 1);
+	return ret;
```

Comment:
1. I think that instead of raising an error, you should handle `nil, err` result in `console.lua`. It wouldn't catch OOM, but we don't care about OOM in Lua, because practically every line of Lua code can fail with OOM.
2. `format_lua_value` suffers from the very same problem. It assumes that `internal.format_lua` throws on error while in fact it returns `nil, err`. Please fix this in the same patch. I understand that it isn't necessary to fix the bugs, but strictly speaking this patch, as well as the next one, are not needed to fix the bugs, either, because the formatting error is ultimately fixed by the third patch AFAICS.

Reply:
Done

Reply:
It turned out that `internal.format_lua` can actually throw, I'll return the `pcall`.